File: gunhoflash/longest-substring-without-repeating-characters.ts
function lengthOfLongestSubstring(s: string): number {
  const L = s.length;
  const S = new Set();
  let answer = 0;
  let left = 0;
  for (let i = 0; i < L; i++) {
    const char = s[i];
    while (S.has(char)) {
      S.delete(s[left++]);
    }
    S.add(char);
    answer = Math.max(answer, i - left + 1);
  }
  return answer;
}

File: gunhoflash/longest-substring-without-repeating-characters.py
class Solution:
  def lengthOfLongestSubstring(self, s: str) -> int:
    L = len(s)
    S = set()
    answer = 0
    left = 0
    for i in range(L):
      char = s[i]
      while char in S:
        S.remove(s[left])
        left += 1
      S.add(char)
      answer = max(answer, i - left + 1)
    return answer
